Frameworks/CoreText/CTLine.mm

```diff
@@ -22,25 +22,47 @@
 #import <CoreText/CTTypesetter.h>
 #import <CoreFoundation/CFString.h>
 
-#include <memory>
+#import <memory>
 #import <algorithm>
 #import <numeric>
 #import <cwchar>
 #import <vector>
 
+// Adds the characters and attributes of a given attributed string to the end of the receiver.
+static inline void __CFAttributedStringAppendAttributedString(CFMutableAttributedStringRef attrString, CFAttributedStringRef appendString) {
+    CFRange replaceRange = { CFAttributedStringGetLength(attrString), 0 };
+    CFAttributedStringReplaceAttributedString(attrString, replaceRange, appendString);
+}
+
+// Inserts the characters and attribute of the given string into the receiver at the given index.
+static inline void __CFAttributedStringInsertAttributedStringOneAttribute(CFMutableAttributedStringRef attrString,
```

Comment:
we keep on creating AttributedStrings and trying to insert them in the for-loop for based on the glyph count, it can be a bottleneck.
If glyph count is usually low, then this isn't needed.

---
In reply to: [112783999](https://github.com/Microsoft/WinObjC/pull/2586#discussion_r112783999) [](ancestors = 112783999)